livedata/src/main/java/io/getstream/chat/android/livedata/extensions/ClientExtensions.kt

```diff
@@ -64,20 +46,29 @@ fun Channel.users(): List<User> {
 }
 
 fun Message.addReaction(reaction: Reaction, isMine: Boolean) {
+
+
     // add to own reactions
     if (isMine) {
+        this.ownReactions = this.ownReactions.toMutableList()
         this.ownReactions.add(reaction)
     }
 
     // add to latest reactions
+    this.latestReactions = this.latestReactions.toMutableList()
     this.latestReactions.add(reaction)
 
     // update the count
     val currentCount = this.reactionCounts.getOrElse(reaction.type) { 0 }
+    // copy the object so livedata's diffutils can notice a change
+    this.reactionCounts = this.reactionCounts.toMutableMap()
```

Comment:
You are mutating `this` (on that case `message`) so the `equals()` method over `message` won't work either